com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/util/ApplicationArchiveExtractorTest.java

```diff
@@ -0,0 +1,59 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.MessageFormat;
+import java.util.stream.Stream;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class ApplicationArchiveExtractorTest {
+
+    private static final String ERROR_SIZE_OF_APP_EXCEEDS_MAX_SIZE_LIMIT = "The size of the application exceeds max size limit \"{0}\"";
+    private static final String SAMPLE_MTAR = "com.sap.mta.sample-1.2.1-beta.mtar";
+    private static final String SAMPLE_FLAT_MTAR = "com.sap.mta.sample-1.2.1-beta-flat.mtar";
+    private static final String SAMPLE_MTAR_WITH_JAR_ENTRY_ABSOLUTE_PATH = "archive-entry-with-absolute-path.mtar";
+    private static final String SAMPLE_MTAR_WITH_JAR_ENTRY_NOT_NORMALIZED_PATH = "archive-entry-with-not-normalized-path.mtar";
+    private static final long MAX_UPLOAD_FILE_SIZE = 1024 * 1024 * 1024l; // 1gb
+
+    public static Stream<Arguments> testExtract() {
+        // @formatter:off
+        return Stream.of(
+            Arguments.of(SAMPLE_MTAR, "web/web-server.zip", null, MAX_UPLOAD_FILE_SIZE),
+            Arguments.of(SAMPLE_FLAT_MTAR, "web/", null, MAX_UPLOAD_FILE_SIZE),
+            Arguments.of(SAMPLE_FLAT_MTAR, "web/", MessageFormat.format(com.sap.cloud.lm.sl.mta.message.Messages.CANNOT_FIND_ARCHIVE_ENTRY, "xxx/web/"), MAX_UPLOAD_FILE_SIZE),
+            Arguments.of(SAMPLE_MTAR_WITH_JAR_ENTRY_ABSOLUTE_PATH, "/web/", MessageFormat.format(com.sap.cloud.lm.sl.cf.core.util.FileUtils.PATH_SHOULD_NOT_BE_ABSOLUTE, "/web/"), MAX_UPLOAD_FILE_SIZE),
+            Arguments.of(SAMPLE_MTAR_WITH_JAR_ENTRY_NOT_NORMALIZED_PATH, "web/", MessageFormat.format(com.sap.cloud.lm.sl.cf.core.util.FileUtils.PATH_SHOULD_BE_NORMALIZED, "web/../asd"), MAX_UPLOAD_FILE_SIZE),
+            Arguments.of(SAMPLE_MTAR, "db/", MessageFormat.format(ERROR_SIZE_OF_APP_EXCEEDS_MAX_SIZE_LIMIT, 200), 200l));
+        // @formatter:on
+    }
+    
+    @ParameterizedTest
+    @MethodSource
+    public void testExtract(String mtar, String fileName, String exceptionMessage, long maxFileUploadSize) throws Exception {
+        InputStream ras = getClass().getResourceAsStream(mtar);
+        try {
+            ApplicationArchiveExtractor streamProducerHandler = new ApplicationArchiveExtractor(ras, fileName, maxFileUploadSize, null) {
+                @Override
+                protected void cleanUp(Path appPath) {
+                    if (appPath == null || !Files.exists(appPath)) {
+                        return;
+                    }
+                    
+                    FileUtils.deleteQuietly(appPath.toFile());
+                }
+            };
+            Path filePath = streamProducerHandler.extract();
+            assertTrue(Files.exists(filePath));

```

Comment:
there is no check to ensure the test fails if it is supposed to throw an exception but executes successfully. See the absolute path test.